user-console/src/main/java/org/pentaho/mantle/client/admin/EmailAdminPanel.java

```diff
@@ -41,12 +42,58 @@ public class EmailAdminPanel extends SimplePanel {
   protected CheckBox useSSLCheckBox;
   protected TextBox fromAddressTextBox;
   protected TextBox fromNameTextBox;
-  protected CheckBox authenticationCheckBox;
   protected TextBox userNameTextBox;
   protected PasswordTextBox passwordTextBox;
   protected Button testButton;
   protected Button saveButton;
   protected VerticalPanel authenticationPanel;
+  protected TextBox tokenUrlTextBox;
+  protected VerticalPanel oauthPanel;
+  protected VerticalPanel passwordPanel;
+  protected ListBox authenticationTypeLB;
+  protected TextBox clientIdTextBox;
+  protected PasswordTextBox clientSecretTextBox;
+  protected TextBox scopeTextBox;
+  protected ListBox grantTypeListBox;
+  protected TextBox refreshTokenTextBox;
+  protected VerticalPanel refreshTokenPanel;
+  protected TextBox authorizationCodeTextBox;
+  protected VerticalPanel authCodePanel;
+  protected TextBox redirectUriTextBox;
+  protected VerticalPanel smtpPanel;
+  /**
+   * The fixed width for all panels in Email admin setup page
+   */
+  private static final String PANEL_WIDTH = "220px";
+
+  /**
+   * The width between items in a HorizontalPanel
+   */
+  private static final String HSPACER_WIDTH = "15px";
+
+  /**
+   * The height between Items in Vertical Panel
+   */
+  private static final String VSPACER_HEIGHT_10 = "10px";
+  /**
+   * The height between Items in Vertical Panel
+   */
+  private static final String VSPACER_HEIGHT_20 = "20px";
+
+  // Authentication Types
+  public static final String AUTH_TYPE_XOAUTH2 = "XOAUTH2";
+  public static final String AUTH_TYPE_BASIC = "basic";
+  public static final String AUTH_TYPE_NO_AUTH = "no_auth";
+
+  // Grant Types
+  public static final String GRANT_TYPE_REFRESH_TOKEN = "refresh_token";
+  public static final String GRANT_TYPE_AUTH_CODE = "authorization_code";
+  public static final String GRANT_TYPE_CLIENT_CREDENTIALS = "client_credentials";
+
+  // Protocols
+  public static final String PROTOCOL_GRAPH_API = "graph_api";
+  public static final String PROTOCOL_SMTP = "SMTP";
+  public static final String PROTOCOL_SMTPS = "SMTPS";
```

Comment:
Why not add these constants on a more "core" type of the object model, such as the interfaces? If you do not want to pollute the interfaces with details which may be seen as specific to implementations, then you could create a Pentaho "internal class" to hold these constants (i.e. a public class but which is documented to be for internal use only).

The way it is, it's already better than it was, but constants are still repeated in multiple files.